### PR TITLE
Added analysis result image viewer

### PR DESCRIPTION
**Pull request by deepsidhu85**
## Description of changes
What did you change in this pull request?  Provide a description of files changed, user interactions changed, etc.  Include how to test your changes.

Added an `Image Output` tab for analysis results on the `Output File Preview` page which will display jpeg and png image previews.

As we don't have a pipeline which outputs image files, testing this out currently requires you to add an image output to a pipeline (upgrading the workflow structure), move an image to the output folder for an Irida pipeline, and updating the path in AnalysisOutputFile for the file. This also requires you to modify the output file map to include this file name. So for now I've included the below screenshot as to what it looks like.

62c0e11c-d9e5-442a-9d2f-a28d182073cf
 

## Related issue
Link to the GitHub issue this pull request addresses using the `#issuenum` format.  If it completes an issue, use `Fixes #issuenum` to automatically close the issue.

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [x] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
~* [ ] Tests added (or description of how to test) for any new features.~
~* [ ] User documentation updated for UI or technical changes.~

### CODE REVIEW

File: src/main/webapp/resources/js/contexts/AnalysisOutputsContext.js [hunk 1]
```diff
@@ -71,7 +78,8 @@ function AnalysisOutputsProvider(props) {
               hasJsonFile: hasJsonFile,
               hasTabularFile: hasTabularFile,
               hasTextFile: hasTextFile,
-              hasExcelFile: hasExcelFile
+              hasExcelFile: hasExcelFile,
```
Comment (joshsadam) [inline src/main/webapp/resources/js/contexts/AnalysisOutputsContext.js:81]: Remember, in es6 this is redundant, you only need:

```js
fileTypes: [
            {
              hasJsonFile,
              hasTabularFile,
              hasTextFile,
              hasExcelFile,
              hasImageFile,
            }
          ]
```

File: src/main/webapp/resources/js/pages/analysis/components/outputs/OutputFilePreview.jsx [hunk 2]
```diff
@@ -20,6 +20,7 @@ import { IconFile } from "../../../../components/icons/Icons";
 const AnalysisTextPreview = React.lazy(() => import("../AnalysisTextPreview"));
 const AnalysisJsonPreview = React.lazy(() => import("../AnalysisJsonPreview"));
 const AnalysisExcelPreview = React.lazy(() => import("../AnalysisExcelPreview"));
+const AnalysisImagePreview = React.lazy(() => import("../AnalysisImagePreview"));
```
Comment (joshsadam) [inline src/main/webapp/resources/js/pages/analysis/components/outputs/OutputFilePreview.jsx:23]: This file is really get big.  Don't do it know, but if you need to add more to it, refactor some helper functions into their own file.

File: src/main/webapp/resources/js/pages/analysis/components/outputs/OutputFilePreview.jsx [hunk 3]
```diff
@@ -152,7 +177,9 @@ export default function OutputFilePreview() {
       if (
         !jsonExtSet.has(output.fileExt) &&
         !tabExtSet.has(output.fileExt) &&
-        !blacklistExtSet.has(output.fileExt)
```
Comment (joshsadam) [inline src/main/webapp/resources/js/pages/analysis/components/outputs/OutputFilePreview.jsx:155]: Isn't this a duplicate set of code from AnalysisOutputsContext?

Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/analysis/components/outputs/OutputFilePreview.jsx:155]: In AnalysisOutputsContext we use this to check if there is a file with type `extension` within the outputs. In the OutputFilePreview we use it to create the preview for that type of file. It is run within a for loop so we need to check if the current output from the array is that type

File: src/main/webapp/resources/js/pages/analysis/components/outputs/OutputFilePreview.jsx (hunk 2 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/analysis/components/outputs/OutputFilePreview.jsx:23]: Yeah I thought about doing that as well. If we need to add another one I will break it up. Thanks

File: src/main/webapp/resources/js/pages/analysis/components/outputs/OutputFilePreview.jsx (hunk 3 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/analysis/components/outputs/OutputFilePreview.jsx:155]: Wouldn't it make more sense to have the context return a function called `getPreviewForFileType`  and have that stored when you first iterate over all the types?

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisAjaxController.java [hunk 4]
```diff
@@ -882,6 +882,39 @@ public void getAjaxDownloadAnalysisSubmissionIndividualFile(@PathVariable Long a
...
+	 * Get an image file associated with a specific {@link AnalysisSubmission} by file name.
+	 *
+	 * @param submissionId {@link Long} id for an {@link AnalysisSubmission}
+	 * @param filename {@link String} filename for an {@link AnalysisOutputFile}
+	 *
+	 * @return {@link String} containing the image file contents as a base64 encoded string.
+	 */
+	@RequestMapping("{submissionId}/image")
+	@ResponseBody
+	public String getImageFile(@PathVariable Long submissionId, String filename){
```
[the remark this answers is not in the document]
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/analysis/AnalysisAjaxController.java]: Updated in a6f973b9319bc3f2d954a416af2062bb5b776ba8

File: src/main/webapp/resources/js/contexts/AnalysisOutputsContext.js (hunk 1 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/contexts/AnalysisOutputsContext.js]: Updated in a6f973b9319bc3f2d954a416af2062bb5b776ba8

File: src/main/webapp/resources/js/pages/analysis/components/outputs/OutputFilePreview.jsx (hunk 3 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/analysis/components/outputs/OutputFilePreview.jsx:155]: Yup totally does. Updated in a6f973b9319bc3f2d954a416af2062bb5b776ba8

**Comment by deepsidhu85** [comment]
> Do you really want to return that message? Maybe return an internationalized message that benefits the user?

I did not want to return that message. Thanks for catching that. Updated in cbf51db393fbb45b195bfb0d8a0dea2fcc2bbced

**Comment by joshsadam** [comment]
Great job, let me know when the tests pass

**Comment by deepsidhu85** [comment]
> Great job, let me know when the tests pass

Thanks!

**Pull request merged**